src/signature_detection/ip_signature.py

```diff
@@ -0,0 +1,32 @@
+from .signature import Signature
+import ipaddress
+import scapy.layers.inet as net
+from scapy.packet import Packet
+from .virustotal_checker import VirusTotalChecker
+from src import run_config
+
+
+class IPSignature(Signature):
+    """
+    This signature detects whether the packet is communicating with a malicious IP address
+    """
+
+    def __init__(self, subnet: str):
+        """
+        Creates this IPSignature
+        :param subnet: The subnet that this router uses in CIDR form, ex. 192.168.1.0/24
+        """
+        self.subnet = ipaddress.ip_network(subnet)
+
+    def __call__(self, packet: Packet) -> bool:
+        if net.IP not in packet:
+            raise Exception
+        ip_layer = packet[net.IP]
+        # check whether source ip is private
+        ip_src = ipaddress.ip_address(ip_layer.src)
+        ip_dst = ipaddress.ip_address(ip_layer.dst)
+        v = VirusTotalChecker(run_config.virustotal_api_key)
```

Comment:
How will this work when the VirusTotalAPIkey is not set and still it's default? Maybe this should be surrounded in a try/except? Thoughts?

Reply:
We can ship one in the config but it will be free and limited in the number of requests/day. I was using my account to test locally but we can make an account with a throwaway email.

Reply:
I would advice against shipping an API key with the package.

In all honesty, if a person is running OpenWrt, we can assume they have minimal technical skills to get an API key and use the config page to input it.